Dockerfile.py3

```diff
@@ -37,37 +40,52 @@ RUN curl https://raw.githubusercontent.com/kadwanev/retry/1.0.1/retry \
 
 ENV USER="user"
 ENV HOME_DIR="/home/${USER}"
-ENV ANDROID_HOME="${HOME_DIR}/.android"
-ENV WORK_DIR="${HOME_DIR}" \
-    PATH="${HOME_DIR}/.local/bin:${PATH}"
+ENV WORK_DIR="${HOME_DIR}/app" \
+    PATH="${HOME_DIR}/.local/bin:${PATH}" \
+    ANDROID_HOME="${HOME_DIR}/.android" \
+    JAVA_HOME="/usr/lib/jvm/java-8-openjdk-amd64"
 
-# install system dependencies
-RUN ${RETRY} apt -y install -qq --no-install-recommends \
-        python3 virtualenv python3-pip python3-venv \
-        wget lbzip2 patch sudo python python-pip \
-    && apt -y autoremove
 
-# build dependencies
-# https://buildozer.readthedocs.io/en/latest/installation.html#android-on-ubuntu-16-04-64bit
+# install system dependencies
 RUN dpkg --add-architecture i386 \
-    && ${RETRY} apt -y update -qq \
+    && ${RETRY} apt -y update -qq > /dev/null \
     && ${RETRY} apt -y install -qq --no-install-recommends \
-        build-essential ccache git python3 python3-dev \
-        libncurses5:i386 libstdc++6:i386 libgtk2.0-0:i386 \
-        libpangox-1.0-0:i386 libpangoxft-1.0-0:i386 libidn11:i386 \
-        zip zlib1g-dev zlib1g:i386 \
-    && apt -y autoremove
-
-# specific recipes dependencies (e.g. libffi requires autoreconf binary)
-RUN ${RETRY} apt -y install -qq --no-install-recommends \
-        libffi-dev autoconf automake cmake gettext libltdl-dev libtool pkg-config \
+    autoconf \
+    automake \
+    build-essential \
+    ccache \
+    cmake \
+    gettext \
+    git \
+    lbzip2 \
+    libffi-dev \
+    libgtk2.0-0:i386 \
+    libidn11:i386 \
+    libltdl-dev \
+    libncurses5:i386 \
+    libpangox-1.0-0:i386 \
+    libpangoxft-1.0-0:i386 \
+    libstdc++6:i386 \
+    libtool \
+    openjdk-8-jdk \
+    patch \
+    pkg-config \
+    python \
+    python-pip \
+    python3 \
+    python3-dev \
+    python3-pip \
+    python3-venv \
+    sudo \
+    unzip \
+    virtualenv \
+    wget \
+    zip \
+    zlib1g-dev \
+    zlib1g:i386 \
```

Comment:
Nice :smile:...far more readable sorted and each dependency in one line :+1:

Reply:
Yes definitely and it's a part of the good practices:
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments